### PR TITLE
update-kernel-versions.sh: Utilize sha256sums.asc

### PR DESCRIPTION
**Pull request by Iglu47**
Partially replace downloading original files to checking with kernel.org checksum autosigner and sha256sums.asc

addition to https://github.com/Frogging-Family/linux-tkg/pull/432
adds a GnuPG dependency, but as far as I understand the current Github workflow should support this:
https://github.com/actions/virtual-environments/blob/ubuntu20/20220508.1/images/linux/Ubuntu2004-Readme.md

### CODE REVIEW

**Comment by Iglu47** [comment]
- I returned the missing single quotes that are needed for `#upcoming_kernel_patch_sha256`.
- for old RC kernels, hashsums are taken directly from the PKGBUILD now (I suppose the probability of finding the correct values there is very high, and if this is not - they will still be overwritten as an obsolete value)
- for the remaining option, I separated the download from the pipeline

I'm not sure if this script works, because with these changes need to test all options...

> Another thing, I believe you fail when the signature verification fails, maybe it's better to retry downloading both the file and signature like 3 times before failing and exiting the script?

well, could probably expect something like this when working with gpg, but if that's going to be a problem, it's probably better to use some other way to manually manage the generation/delivery of keyring.gpg (especially since they don't update very often).

**Comment by AdelKS** [comment]
> for old RC kernels, hashsums are taken directly from the PKGBUILD now (I suppose the probability of finding the correct values there is very high, and if this is not - they will still be overwritten as an obsolete value)

Very good idea

> for the remaining option, I separated the download from the pipeline

:+1: 

> I'm not sure if this script works, because with these changes need to test all options...

The only option I did not test is going from rc to release. And that one is hard to test (without some extra wokr), we just need to wait for the actual release and we'll see xD

> well, could probably expect something like this when working with gpg, but if that's going to be a problem, it's probably better to use some other way to manually manage the generation/delivery of keyring.gpg (especially since they don't update very often).

I wasn't talking about that, I will comment on the actual line

File: update-kernel-versions.sh
Comment (AdelKS) [inline update-kernel-versions.sh:58]: Here, if it fails, I think it's better to re-download the shasum file and its signature again, then test again. 

But you know what ? Let's see if it will actually fail in the upcoming updates. If it fails too frequently, we can do the change. Also, the CI run is run every six hours, so if it fails, in the next run it probably won't. But people will have to wait 6 more hours :fearful: xD

**Comment by AdelKS** [comment]
I tested your updated script on 599eb50 (by checking out that commit and simply copying your updated script as `update-kernel-versions-new.sh`. And I get this issue:

```shell
adel@leyantoo /media/personal/Documents/Gethings/linux-tkg                                                                                                      [21:39:21]
> $ ./update-kernel-versions-new.sh                                                                                                                           [±599eb50 ●]
gnupg directory does not exist
Making sure we have all the necessary keys
pub   rsa4096 2013-01-24 [SC]
      B8868C80BA62A1FFFAF5FDA9632D3A06589DA6B1
uid           [ unknown] Kernel.org checksum autosigner <autosigner@kernel.org>

pub   rsa4096 2011-09-23 [SC]
      647F28654894E3BD457199BE38DBBDC86092693E
uid           [ unknown] Greg Kroah-Hartman <gregkh@kernel.org>
sub   rsa4096 2011-09-23 [E]

pub   rsa2048 2011-09-20 [SC]
      ABAF11C65A2970B130ABE3C479BE3E4300411886
uid           [ unknown] Linus Torvalds <torvalds@kernel.org>
sub   rsa2048 2011-09-20 [E]

Downloading the checksums file for linux-v5.x
Verifying the v5.x checksums file
gpg: WARNING: no command supplied.  Trying to guess what you mean ...
gpg: Signature made Sun 15 May 2022 08:33:07 PM CEST
gpg:                using RSA key 632D3A06589DA6B1
gpg: Good signature from "Kernel.org checksum autosigner <autosigner@kernel.org>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: B886 8C80 BA62 A1FF FAF5  FDA9 632D 3A06 589D A6B1
current version on repository 5.18.rc5 -> 5
upstream version 5.18-rc7 -> 7
Newer upstream
Updating rc kernel version in linux-tkg-config/prepare
Downloading the GZ tarball for linux-5.18-rc5
Warning: Binary output can mess up your terminal. Use "--output -" to tell 
Warning: curl to output it to your terminal anyway, or consider "--output 
Warning: <FILE>" to save to a file.
FAILED to download the linux-5.18-rc5.tar.gz
```

**Comment by Iglu47** [comment]
> Downloading the GZ tarball for linux-5.18-rc5
> Warning: Binary output can mess up your terminal. Use "--output -" to tell 
> Warning: curl to output it to your terminal anyway, or consider "--output 
> Warning: <FILE>" to save to a file.
> FAILED to download the linux-5.18-rc5.tar.gz

thanks, I didn't notice how it was leaked to the public version. fixed now.

File: update-kernel-versions.sh
Comment (Iglu47) [reply update-kernel-versions.sh:58]: Hmm, Yea, I would come back to this part if there will be any often problem with this. And maybe would make the script re-invoke via Github workflow somehow. So that all exits from the script with error 3 are simply given a second chance by re-running the script.

Comment (Iglu47) [reply update-kernel-versions.sh:58]: about 6 hours: Also, I have not yet found a way to replace the scheduled launch of the script. I thought there would be some ready-made option for Github, but maybe something can be done with tracking release tags, for example via email or something, and then triggering an event by writing a comment in the issue/discossion.

Comment (AdelKS) [reply update-kernel-versions.sh:58]: Let's merge this PR then we can think of this improvement in another

**Comment by AdelKS** [comment]
Looks all good to me! We can merge this already @Tk-Glitch

**Review by AdelKS — APPROVED** [review]

**Comment by AdelKS** [comment]
Unfortunately I still do not have write access to merge PRs so we have to wait for TKG.

**Comment by Iglu47** [comment]
@AdelKS I changed the command for finding `old_kernel_shasum` to be shorter and more correct.

**Comment by Tk-Glitch** [comment]
Looks good to me. Let's give it a spin!
Thank you Iglu47 for diving into it, and thank you Adel for reviewing sand testing the changes!
🐸 ❤️

**Pull request merged**
